File: GC.WebSpace/ClientApp/domain/gardens/gardeners/gardener.ts
export class Gardener {
    constructor(
        public readonly id: string,
        public readonly firstName: string,
        public readonly middleName: string | null,
        public readonly lastName: string,
        public readonly sectorId: string
    ) { }

    public initials(): string {
        return `${this.lastName} ${this.firstName[0]}.${this.middleName == null ? '' : ` ${this.middleName[0]}.`}`;
    }

    public fullName(): string {
        return `${this.lastName} ${this.firstName} ${this.middleName == null ? '' : ` ${this.middleName}`}`;
    }
}

export namespace Gardener {
    export function toModel(value: any) {
        return new Gardener(value.id, value.firstName, value.middleName, value.lastName, value.sectorId);
    }

    export function toModels(values: any[]) {
        return values.map(toModel);
    }
}

File: GC.Domain/Gardens/Gardener.cs
using GC.Domain.Persons;
using System;

namespace GC.Domain.Gardens
{
    public class Gardener : FromPerson
    {
        public Guid Id { get; }
        public String FirstName { get; }
        public String? MiddleName { get; }
        public String LastName { get; }
        public Guid SectorId { get; }

        public String FullName => $"{LastName} {FirstName} {MiddleName ?? ""}";
        public String Initials => $"{LastName} {FirstName[0]}. {$"{MiddleName}." ?? ""}";

        public Gardener(Guid id, String firstName, String? middleName, String lastName, Guid sectorId)
        {
            Id = id;
            FirstName = firstName;
            MiddleName = middleName;
            LastName = lastName;
            SectorId = sectorId;
        }
    }
}
